source/fe/fe_raviart_thomas.cc

```diff
@@ -248,6 +248,173 @@ FE_RaviartThomas<dim>::initialize_support_points(const unsigned int deg)
+  // y-direction running from 0 to n-1.  to extract i and j we can use
+  // i=dof_index%n and j=dof_index/n. The indices i and j can then be used to
+  // compute the offset.
+
+  // Example: if the switches are (true | true | true) that means we flip the
+  // face first by + 90 degree(counterclockwise) then by another +180
+  // degrees but we do not flip it since the face has standard
+  // orientation. The flip axis is the diagonal from the lower left to the upper
+  // right corner of the face. With these flags the configuration above becomes:
+  //  ___________
+  // |           |
+  // |  8  7  6  |
+  // |           |
+  // |  5  4  3  |
+  // |           |
+  // |  2  1  0  |
+  // |___________|
+  //
+  // This is exactly what is realized by the formulas implemented below. Note
+  // that the necessity of a permuattion depends on the three flags.
+
+  // There is also a pattern for the sign change of the mapped face_dofs
+  // depending on the switches. In the above example it would be
+  //  ___________
+  // |           |
+  // |  +  +  -  |
+  // |           |
+  // |  +  +  -  |
+  // |           |
+  // |  +  +  -  |
+  // |___________|
+  //
+
+  for (unsigned int local_face_dof = 0;
+       local_face_dof < this->n_dofs_per_quad(face_no);
+       ++local_face_dof)
+    {
+      // Row and column
+      unsigned int i = local_face_dof % n, j = local_face_dof / n;
+
+      // We have 8 cases that are all treated the same way. Note that the
+      // corresponding case to case_no is just its binary representation.
+      // The above example of (false | true | true) would be case_no=3
+      for (unsigned int case_no = 0; case_no < 8; ++case_no)
+        {
+          // Get the binary representation of the case
+          const bool face_orientation = (((case_no / 4) % 2) == 1);
+          const bool face_flip        = (((case_no / 2) % 2) == 1);
+          const bool face_rotation    = ((case_no % 2) == 1);
+
+          if (((!face_orientation) && (!face_rotation)) ||
+              ((face_orientation) && (face_rotation)))
+            {
+              // We flip across the diagonal
+              // This is the local face dof offset
+              this->adjust_quad_dof_index_for_face_orientation_table[face_no](
+                local_face_dof, case_no) = j + i * n - local_face_dof;
+            }
+          else
+            {
+              // Offset is zero
+              this->adjust_quad_dof_index_for_face_orientation_table[face_no](
+                local_face_dof, case_no) = 0;
+            } // if face needs dof permutation
+
+          // Get new local face_dof by adding offset
+          unsigned int new_local_face_dof =
+            local_face_dof +
+            this->adjust_quad_dof_index_for_face_orientation_table[face_no](
+              local_face_dof, case_no);
+          // compute new row and column index
+          i = new_local_face_dof % n;
+          j = new_local_face_dof / n;
```

Comment:
I think you're overwriting what you already computed before here. If you don't need the values of `i` and `j` previously computed, then just move the variable declarations to here.

Reply:
The variables are overwritten in each iteration and used. See above comment. The style is maybe not good imo and I will think about a better design. But the tests indicate that it works.

Reply:
It's not terribly important, but as a general rule, move declarations of variables as far down in the function as you can, into the innermost scope possible. This way it becomes easier to reason about questions like "Where does this variable live?" or "Does it carry state from one loop iteration to the next?"